### ISSUE TITLE
Analyzing TIFF files with time frames < 500

### ISSUE THREAD

**Issue by ViciousPlants** [nel-lab/mesmerize-core#323]
After installing mesmerize-core and mesmerize-vis I was able to run "mcorr_cnmf.ipynb" with the "Sue_2x_3000_40_-46.tif" example. However, when loading my own TIFF files, running caiman gives error at the step of "Computing correlation image". After playing around with the number of frames (t) in the "Sue_2x_3000_40_-46.tif" example, I discovered that the same type of error is encountered whenever there is frame < 500. (_Note, changing "num_frames_split" in the mcorr_params1 based on what I can find online does not help with this issue_)

Below is the traceback error for "Sue_2x_3000_40_-46.tif" with 499 frames:
![Mes_err](https://github.com/user-attachments/assets/31a3a039-17e3-40ec-a351-dd1b0d907be1)

This is my first time trying to analyze calcium imaging data and although the user guides and API have much technical information, I am unsure how to link them to biological data/interpretation. Is there any reason why the input images/movies must be >= 500 frames?

 - OS: WIndows 10
 - mesmerize-core version: 0.4.0. install via pip with python 3.10
 - caiman version: 1.11.3
- RAM: 128 GB
- CPU: Intel i9-10940X 
- GPU: NVIDIA RTX A4000

**Comment by ethanbb** [comment]
I could be wrong but I don't think this is intentional...`local_correlations_movie_offline` (in base CaImAn, not mesmerize-core) does not behave correctly if the window size is less than the number of frames; it should raise a more informative error, but it hasn't been implemented. Right now the window size in the call in mesmerize is hard-coded to 1000. (Incidentally, if the number of frames is between 500 and 1000, it silently does the wrong thing.) 

I think the best way to work around is to use min(1000, T) for the window size @kushalkolar? This footgun should really also be fixed upstream though.

**Comment by kushalkolar** [comment]
I don't know if this will be fixed, easiest way for you to move forward is to use your own fork and change the hardcoded values here: https://github.com/nel-lab/mesmerize-core/blob/master/mesmerize_core%2Falgorithms%2Fmcorr.py#L95-L96

> I am unsure how to link them to biological data/interpretation.

What specifically are you unsure about?

**Comment by kushalkolar** [comment]
It's the time dimension. Where did you make the change, if you have a local fork you have to pip install it in place. From the traceback it shows it's using the mescore in the environment.

**Comment by ethanbb** [comment]
Yeah to add to this, sorry it was unclear, the "window" has nothing to do with X and Y so you don't have to truncate your movie or do anything else to it. It's just the length of the window in time that the local correlation algorithm uses, which can't be longer than the number of frames in your data.

**Comment by ViciousPlants** [comment]
> It's the time dimension. Where did you make the change, if you have a local fork you have to pip install it in place. From the traceback it shows it's using the mescore in the environment.

I installed mescore with mamba by creating a mescore env. I changed the code within the python script where it was installed in the mescore environment "C:\Users\name\miniconda3\envs\mescore\Lib\site-packages\mesmerize_core\algorithms" 

Wouldn't pip install create a new script with the original values?

**Comment by kushalkolar** [comment]
You want to do an [inplace install](https://stackoverflow.com/questions/35064426/when-would-the-e-editable-option-be-useful-with-pip-install) with a local fork where you maek your changes

```python
git clone https://github.com/nel-lab/mesmerize-core.git
# make your changes
cd mesmerize-core
pip install -e .
```

**Comment by ViciousPlants** [comment]
Update: I have successfully changed the mcorr.py and processed my data. 

Follow-up: My actual data is 3D +time and the demo only have 2D data. Do you have any recommendations for the analysis of 3D data and options to tinker with?

**Comment by kushalkolar** [comment]
mesmerize-viz only supports 2D visualizations, you'll have to use the fastplotlib `ImageWidget` directly: https://github.com/fastplotlib/fastplotlib/blob/v0.1.0.a16/examples/notebooks/image_widget.ipynb

**Comment by ethanbb** [comment]
Yes mesmerize-core (but not -viz) should support 3D motion correction and CNMF, as long as you are using CaImAn version 1.11.2 or later. Give it a try and let me know if you run into any problems.

**Comment by ethanbb** [comment]
I've never run into that error, unfortunately. I looked through the code a bit and it seems like with border_nan = 'copy', it would be unlikely for the template to end up containing NaNs unless the input image had NaNs, but I could be missing something. You could either try to step through carefully with a debugger and see where you start getting NaNs, or ask on the main CaImAn discussion board (https://github.com/flatironinstitute/CaImAn/discussions)
